executor/show_test.go

```diff
@@ -168,13 +168,40 @@ func (s *testSuite) TestShow(c *C) {
 	// Test show create table with AUTO_INCREMENT option
 	// for issue https://github.com/pingcap/tidb/issues/3747
 	tk.MustExec(`drop table if exists show_auto_increment`)
-	tk.MustExec(`create table show_auto_increment (id int) auto_increment=4`)
+	tk.MustExec(`create table show_auto_increment (id int key auto_increment) auto_increment=4`)
 	tk.MustQuery(`show create table show_auto_increment`).Check(testutil.RowsWithSep("|",
 		""+
 			"show_auto_increment CREATE TABLE `show_auto_increment` (\n"+
-			"  `id` int(11) DEFAULT NULL\n"+
+			"  `id` int(11) NOT NULL AUTO_INCREMENT,\n"+
+			"  PRIMARY KEY (`id`)\n"+
 			") ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_bin AUTO_INCREMENT=4",
 	))
+	// for issue https://github.com/pingcap/tidb/issues/4678
+	tk.MustExec("insert into show_auto_increment values(20)")
+	tk.MustQuery(`show create table show_auto_increment`).Check(testutil.RowsWithSep("|",
+		""+
+			"show_auto_increment CREATE TABLE `show_auto_increment` (\n"+
+			"  `id` int(11) NOT NULL AUTO_INCREMENT,\n"+
+			"  PRIMARY KEY (`id`)\n"+
+			") ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_bin AUTO_INCREMENT=5021",
```

Comment:
Plz add some comment here, about why this is 5011, not 11, autoid step is 5000 in `meta/autoid/autoid.go:30`

Reply:
or, we can exported the step in autoid, and use it in here.